### ISSUE TITLE
Improve MasterDetailLayout samples for a11y

### ISSUE THREAD

**Issue by rolfsmeds** [vaadin/docs#4209]
Remember to provide instructions for screen reader support:
- Mark up master and especially details as `<section>` / `role="region"`
- Use appropriate headings for each area (detail level +1 level compared to master; maybe also `aria-labelledby` should be used to associate each area with their titles – not sure if necessary)
- Maybe show how to generate announcement when contents of details changes?
